eo-parser/src/main/java/org/eolang/parser/StrictXmir.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    @Override
+    public List<XML> nodes(final String query) {
+        return this.xml.nodes(query);
+    }
+
+    @Override
+    public XML registerNs(final String prefix, final Object uri) {
+        return this.xml.registerNs(prefix, uri);
+    }
+
+    @Override
+    public XML merge(final NamespaceContext context) {
+        return this.xml.merge(context);
+    }
+
+    @Override
+    @Deprecated
+    public Node node() {
+        throw new UnsupportedOperationException("deprecated");
+    }
+
+    @Override
+    public Node inner() {
+        return this.xml.inner();
+    }
+
+    @Override
+    public Node deepCopy() {
+        return this.xml.deepCopy();
+    }
+
+    @Override
+    public Collection<SAXParseException> validate() {
+        return this.xml.validate();
+    }
+
+    @Override
+    public Collection<SAXParseException> validate(final XML schema) {
+        return this.xml.validate(schema);
+    }
+
+    /**
+     * Here, we check the location of the XSD in the XML
+     * and replace with a new one, if necessary.
+     * @param xml Original XML
+     * @return New XML with the same node
+     */
+    private static XML reset(final XML xml) {
+        final Node node = xml.inner();
+        final List<String> location = xml.xpath("/program/@xsi:noNamespaceSchemaLocation");
+        if (!location.isEmpty()) {
+            String uri = location.get(0);
+            if (uri.startsWith("http")) {
+                uri = String.format(
+                    "file:///%s",
+                    StrictXmir.download(
+                        uri,
+                        Paths.get("target/xsd").resolve(
+                            uri.substring(uri.lastIndexOf('/') + 1)
+                        )
+                    ).toString().replace("\\", "/")
+                );
+            }
+            new Xembler(
+                new Directives().xpath("/program").attr(
+                    "noNamespaceSchemaLocation xsi http://www.w3.org/2001/XMLSchema-instance",
+                    uri
+                )
+            ).applyQuietly(node);
+        }
+        return new XMLDocument(node);
+    }
+
+    /**
+     * Download URI from Internet and save to file.
+     * @param uri The URI
+     * @param path The file
+     * @return Where it was saved
+     */
+    private static File download(final String uri, final Path path) {
+        final File abs = path.toFile().getAbsoluteFile();
+        if (!abs.exists()) {
+            if (abs.getParentFile().mkdirs()) {
+                Logger.debug(StrictXmir.class, "Directory for %[file]s created", path);
+            }
+            try {
+                Files.write(
+                    path,
+                    new IoCheckedBytes(
+                        new BytesOf(new InputOf(new URI(uri)))
+                    ).asBytes()
+                );
+            } catch (final IOException | URISyntaxException ex) {
+                throw new IllegalArgumentException(ex);
```

Comment:
@yegor256 [NIT] It's a bad idea by a several reasons:
1) Exception without context: https://www.yegor256.com/2015/12/01/rethrow-exceptions.html
2) Grouped exceptions: https://www.yegor256.com/2022/08/30/dont-group-exception-catchers.html

Reply:
@volodya-lombrozo fixed